src/libraries/System.Security.Cryptography.Encoding/tests/Cbor/CborReader.cs

```diff
@@ -135,9 +139,25 @@ public CborReaderState Peek()
                 CborMajorType.Array => CborReaderState.StartArray,
                 CborMajorType.Map => CborReaderState.StartMap,
                 CborMajorType.Tag => CborReaderState.Tag,
-                CborMajorType.Special => CborReaderState.Special,
+                CborMajorType.Special => MapSpecialValueTagToReaderState(initialByte.AdditionalInfo),
```

Comment:
I'm concerned this is going to be an evolution problem.

If an RFC is released that defines special item 28, then the version of this library released at the time will report it as `SpecialValue`, because it's a "future unknown".  Early adopters build against that library, look for SpecialValue, and do the work of interpreting it on their own.  They release as shared libraries (not applications), and everything's hunky-dory.  Then we release a new version of the library that supports SpecialType28, their case statement stops matching, and the other library (and application hosting it) explode.

That might just mean that `SpecialValue` needs to be `UnknownSpecialValue`, and we need an 'IsSpecialValue` method or property somewhere; then document that it's an unstable return value and callers are encouraged to use the predicate instead to determine fallback conditions.

Reply:
I was having similar thoughts about this. In the end, I went for specializing the reader state only for null and boolean special values, since my expectation is that they will be used frequently (particularly if mapping from json docs). For instance, the `Undefined` special value is not mapped to a corresponding reader state. Note that floating-point representations, while they do share the same major type as special values, are not technically special values.

I'd be inclined to say that null and boolean representations should be the only exceptions, and we should not be changing the reader state for any current or future special values to be added.